gammapy/image/maps.py

```diff
@@ -242,25 +243,38 @@ def coordinates(self, coord_type='skycoord', origin=0, mode='center'):
         else:
             raise ValueError('Invalid mode to compute coordinates.')
 
-        if coord_type == 'pix':
-            return x, y
-        else:
-            coordinates = pixel_to_skycoord(x, y, self.wcs, origin)
-            if coord_type == 'skycoord':
-                return coordinates
-            elif coord_type == 'galactic':
-                l = coordinates.galactic.l.wrap_at('180d')
-                b = coordinates.galactic.b
-                return l, b
-            else:
-                raise ValueError("Not a valid coordinate type. Choose either"
-                                 " 'pix' or 'skycoord'.")
+        return x, y
+
+    def coordinates(self, origin=0, mode='center'):
+        """
+        Sky coordinate images.
+
+        Parameters
+        ----------
+        origin : {0, 1}
+            Pixel coordinate origin.
+        mode : {'center', 'edges'}
+            Return coordinate values at the pixels edges or pixel centers.
+
+        Returns
+        -------
+        coordinates : tuple or `~astropy.coordinates.SkyCoord`
+            Position on the sky. Can be either an instance of
+            `~astropy.coordinates.SkyCoord` or a tuple of `~numpy.ndarray`
+            of the form (lon, lat) or (ra, dec), depending on the WCS
+            transformation that is set for the sky map.
+        """
+        x, y = self.pix_coordinates(mode=mode)
+        coordinates = pixel_to_skycoord(x, y, self.wcs, origin)
+        return coordinates
 
     def solid_angle(self):
         """
         Solid angle image
         """
-        xsky, ysky = self.coordinates(coord_type='galactic', mode='edges')
+        coordinates = self.coordinates(mode='edges')
+        xsky = coordinates.galactic.l.wrap_at('180d')
```

Comment:
You shouldn't force Galactic coordinates here and in other places in this PR.

What you want is just the lon / lat in whatever coordinate frame the image is (e.g. galactic or icrs or ...).

So instead of calling:
```
xsky = coordinates.galactic.l.wrap_at('180d')
ysky = coordinates.galactic.b
```
you should call
```
xsky = coordinates.data.lon
ysky = coordinates.data.lat
```
and then you'll get the lon / lat in the coordinate frame of the image

There will be very few cases in Gammapy where we want to force the frame to `galactic` or `icrs`.